lib/datadog/opentelemetry/metrics.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+module Datadog
+  module OpenTelemetry
+    module Metrics
+      module Initializer
+        module_function
+
+        def initialize!(settings)
+          return false unless settings.opentelemetry.metrics.enabled
+
+          unless defined?(::OpenTelemetry::SDK)
+            Datadog.logger.warn('OpenTelemetry metrics requested but opentelemetry-sdk gem is not available')
+            return false
+          end
+
+          unless defined?(::OpenTelemetry::SDK::Metrics)
+            Datadog.logger.warn('OpenTelemetry metrics requested but opentelemetry-metrics-sdk gem is not available')
+            return false
+          end
+
+          configure_metrics_sdk(settings)
+          true
+        rescue StandardError => e
+          Datadog.logger.error("Failed to initialize OpenTelemetry metrics: #{e.message}")
+          Datadog.logger.debug(e.backtrace.join("\n")) if Datadog.configuration.diagnostics.debug
+          false
+        end
+
+
+        private
+
+        module_function
+
+        def configure_metrics_sdk(settings)
+          require 'opentelemetry/sdk'
+          require 'opentelemetry-metrics-sdk'
+
+          current_provider = ::OpenTelemetry.meter_provider
+          if current_provider.is_a?(::OpenTelemetry::SDK::Metrics::MeterProvider)
+            current_provider.shutdown
+          end
+
+          resource = create_resource(settings)
+          provider = ::OpenTelemetry::SDK::Metrics::MeterProvider.new(resource: resource)
+          configure_metric_reader(provider, settings)
+          ::OpenTelemetry.meter_provider = provider
+        end
+
+        def create_resource(settings)
+          resource_attributes = {'host.name' => Datadog::Core::Environment::Socket.hostname}
+
+          settings.tags&.each do |key, value|
```

Comment:
Should `Datadog::Core::Environment::Socket.hostname` override the user-provided tag with key `host.name`? Today the user-provided tag overrides `Datadog::Core::Environment::Socket.hostname`. I just want to make sure that this is intentional.